src/coreclr/vm/jitinterface.cpp

```diff
@@ -3369,14 +3369,24 @@ int CEEInfo::appendClassName(_Outptr_result_buffer_(*pnBufLen) char16_t** ppBuf,
     nLen = (int)wcslen(szString);
     if (*pnBufLen > 0)
     {
-    wcscpy_s((WCHAR*)*ppBuf, *pnBufLen, szString );
-    (*ppBuf)[(*pnBufLen) - 1] = W('\0');
-    (*ppBuf) += nLen;
-    (*pnBufLen) -= nLen;
+        // Copy as much as will fit.
+        WCHAR* pBuf = (WCHAR*)*ppBuf;
+        int nLenToCopy = min(*pnBufLen, nLen + /* null terminator */ 1);
+        for (int i = 0; i < nLenToCopy; i++)
```

Comment:
Why not use `memcpy` instead?

Reply:
Also shouldn't it be `i < (nLenToCopy - 1)`?

Reply:
We could use memcpy; I wrote it this way so it looks more similar to the crossgen2 version, although they're not identical, so it's a minor point.